cpp/src/sparse/cluster/cluster_solvers.cuh

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2019-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __CLUSTER_SOLVERS_H
+#define __CLUSTER_SOLVERS_H
+
+#pragma once
```

Comment:
Do we need both `#pragma once` and the header define `#define __CLUSTER_SOLVERS_H`? Typically those are used for the same purpose and only one is needed. I would favor `#pragma once` between the two.

Reply:
This stuff was copied from raft so that we can move towards finishing the migration. We will continue to improve in cuVSz the grammian stuff specifically is complicated right now- the APIs need a significant update (which means removing the complex classes altogether and using free functions to match the rest of cuVs). we aren't going to bother making those changes yet because we want minimal impact to cuml up front (eg we just want to be able to update the namespaces in cuml for 25.02).

Reply:
Also this whole cluster solvers API is going away. The spectral clustering and embedding APIs in general do not match the rest of cuVS (nor raft).